### PR TITLE
RSDK-3975: Single line encoder double counts on a single line physical encoder

### PR DESCRIPTION
**Pull request by martha-johnston**
This PR fixes the issue where single encoders were double counting ticks. Now it will only increase the encoder position if the tick value has changed. Tested on a viam rover with a pi and 996 ticks was equivalent to 1 revolution as expected, and as seen with an Arduino

### CODE REVIEW

File: components/encoder/single/single_encoder.go [hunk 1]
```diff
@@ -205,7 +206,15 @@ func (e *Encoder) Start(ctx context.Context) {
...
-					atomic.AddInt64(&e.position, dir)
+					tick, err := e.I.Value(ctx, nil)
+					if err != nil {
+						e.logger.Error(err)
+						return
+					}
+					if tick != lastTick {
+						atomic.AddInt64(&e.position, dir)
+					}
+					lastTick = tick
```
Comment (randhid) [inline components/encoder/single/single_encoder.go:217]: beautiful. test it out on a motor with a quadrature encoder (on my desk) and the viam-rover to make sure this still works in both cases.

**Review by randhid — APPROVED** [review]
One more hardware test and we can get it merged in.

File: components/encoder/single/single_encoder.go (hunk 1 above)
Comment (randhid) [reply components/encoder/single/single_encoder.go:217]: do we need a mutex to protect any of this? just noticing.

Comment (martha-johnston) [reply components/encoder/single/single_encoder.go:217]: hmm not sure. it didn't seem to have any issues but maybe when reading the tick value on line 209?

**Pull request merged**